content/webapp/components/EventsByMonth/group-event-utils.ts

```diff
@@ -31,8 +31,15 @@ export function parseLabel(label: string): YearMonth {
   return { year, month };
 }
 
+/** Returns the UTC start of a given month.
+ *
+ * Note: we use UTC to ensure consistent behaviour, and because our comparison
+ * functions (e.g. isSameMonth) use UTC dates also.
+ */
 export function startOf({ year, month }: YearMonth): Date {
-  return new Date(year, month, 1);
+  return new Date(
+    `${year}-${(month + 1).toString().padStart(2, '0')}-01T01:01:01Z`
+  );
```

Comment:
This is the sort of thing that makes me think we should be looking at `date-fns` rather than implementing our own wrangling logic

Reply:
yeah